src/Features/Core/Portable/IntroduceUsingStatement/AbstractIntroduceUsingStatementCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,291 @@
+            TextSpan span,
+            CancellationToken cancellationToken)
+        {
+            var (declarationStatement, localVariable) = await FindDisposableLocalDeclaration(document, span, cancellationToken).ConfigureAwait(false);
+
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+            if (!(declarationStatement.Parent is TBlockSyntax parent))
+            {
+                throw new NotImplementedException();
+            }
+
+            var syntaxFactsService = document.GetLanguageService<ISyntaxFactsService>();
+
+            var lastUsageStatement = FindSiblingStatementContainingLastUsage(
+                declarationStatement,
+                localVariable,
+                semanticModel,
+                document.GetLanguageService<ISyntaxFactsService>(),
+                cancellationToken);
+
+            var parentStatementsList = GetStatements(parent);
+
+            var declarationStatementIndex = parentStatementsList.IndexOf(declarationStatement);
+
+            var lastUsageStatementIndex = lastUsageStatement is null
+                ? -1
+                : parentStatementsList.IndexOf(lastUsageStatement, declarationStatementIndex + 1);
+
+            var statementsToSurround = GetStatementsToSurround(parentStatementsList, declarationStatementIndex, lastUsageStatementIndex);
+
+
+            var trailingTrivia = SplitTrailingTrivia(declarationStatement, syntaxFactsService);
+
+            var usingStatement =
+                CreateUsingStatement(
+                    declarationStatement.WithoutTrivia(),
+                    trailingTrivia.sameLine,
+                    statementsToSurround)
+                    .WithLeadingTrivia(declarationStatement.GetLeadingTrivia())
+                    .WithTrailingTrivia(trailingTrivia.followingLines);
+
+            var newParent = WithStatements(
+                parent,
+                ReplaceRange(
+                    parentStatementsList,
+                    index: declarationStatementIndex,
+                    count: 1 + statementsToSurround.Count,
+                    newNodes: new[] { usingStatement }));
+
+            return document.WithSyntaxRoot(root.ReplaceNode(parent, newParent));
+        }
+
+        private static (SyntaxTriviaList sameLine, SyntaxTriviaList followingLines) SplitTrailingTrivia(SyntaxNode node, ISyntaxFactsService syntaxFactsService)
+        {
+            var trailingTrivia = node.GetTrailingTrivia();
+
+            for (var i = 0; i < trailingTrivia.Count; i++)
+            {
+                if (syntaxFactsService.IsEndOfLineTrivia(trailingTrivia[i]))
+                {
+                    return (
+                        sameLine: new SyntaxTriviaList(trailingTrivia.Take(i)),
+                        followingLines: new SyntaxTriviaList(trailingTrivia.Skip(i)));
```

Comment:
instead of 'followingLines', it sounds like this should just be `endOfLineTrivias`